### PR TITLE
fix for DNS issue in OpenShift

### PR DESCRIPTION
**Pull request by onong**
NetworkManager wipes out DNS config when it sees that the interface has vanished/changed. The fix appends "dns=none" in the NetworkManager.conf file which prevents this calamity.

### CODE REVIEW

File: vpp-manager/startup/startup.go [hunk 1]
```diff
@@ -115,13 +115,21 @@ func parseEnvVariables(params *config.VppManagerParams) (err error) {
 		hooks.RegisterBashHook(hooks.BEFORE_IF_READ, conf)
 	}
 
+	// NetworkManager wipes out DNS config when the uplink interface
+	// is moved. We counter it by telling it to leave the DNS config alone.
+	hooks.RegisterBashHook(hooks.BEFORE_VPP_RUN, hooks.FIX_DNS_SCRIPT)
```
Comment (sknat) [inline vpp-manager/startup/startup.go:120]: Maybe that'd be easier with renaming `network_restart.sh` : `default_hook.sh` and passing the hookpoint when calling it.
That way we can start the script with
````bash
if [ $1 = "BEFORE_VPP_RUN" ]; then 
   ...
elif ....
````
and have everything in a single glance.

File: vpp-manager/startup/startup.go [hunk 2]
```diff
@@ -115,13 +115,21 @@ func parseEnvVariables(params *config.VppManagerParams) (err error) {
...
+
 	// Add default hook if none specified
 	for _, hookName := range []string{hooks.VPP_RUNNING, hooks.VPP_DONE_OK, hooks.VPP_ERRORED} {
 		if hooks.HookCount(hookName) == 0 {
 			hooks.RegisterBashHook(hookName, hooks.DEFAULT_RESTART_SCRIPT)
```
Comment (sknat) [inline vpp-manager/startup/startup.go:125]: Having a single script would also allow user override with this function (we should probably add `BEFORE_VPP_RUN` to the lot of hooks having our script as the default)

**Review by sknat — COMMENTED** [review]
overall lgtm,
Just a few comments regarding how defaults behave / where the logic goes

File: vpp-manager/hooks/default_hook.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+
+HOOK="$0"
```
Comment (sknat) [inline vpp-manager/hooks/default_hook.sh:3]: Shouldn't it be `$1` ? (iirc `$0` is the script name)

File: vpp-manager/startup/startup.go (hunk 1 above)
Comment (onong) [reply vpp-manager/startup/startup.go]: Great idea. Thanks.

File: vpp-manager/hooks/default_hook.sh (hunk 3 above)
Comment (sknat) [reply vpp-manager/hooks/default_hook.sh:3]: Right, my bad 😕 
I didn't recall we were passing it inline

**Review by sknat — APPROVED** [review]

**Pull request merged**